### PR TITLE
[IOTDB-2396] Compatible with 0.12 on language structure

### PR DESCRIPTION
**Pull request by liuminghui233**
Compatible with 0.12 on language structure:

 - Support . in the node name of the path
 - Usage of the single and double quotes

For more details, please refer to `Syntax Conventions.md`

### CODE REVIEW

**Review by HTHou — APPROVED** [review]
LGTM

File: docs/zh/UserGuide/Appendix/Keywords.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+-->
+
+# 关键字和保留字
+
+以下为 IoTDB 0.13 版本的关键字，其中保留字用 **(R)** 标志。
+
+- 一般关键字
+    - ADD
+    - A F T E R
```
Comment (qiaojialin) [inline docs/zh/UserGuide/Appendix/Keywords.md:28]:
```suggestion
    - AFTER
```
？

File: docs/UserGuide/Appendix/Keywords.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+# Keywords and Reserved Words
```
Comment (SteveYurongSu) [inline docs/UserGuide/Appendix/Keywords.md:22]: Add it to config.js

File: docs/zh/UserGuide/Appendix/Keywords.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+# 关键字和保留字
+
+以下为 IoTDB 0.13 版本的关键字，其中保留字用 **(R)** 标志。
```
Comment (SteveYurongSu) [inline docs/zh/UserGuide/Appendix/Keywords.md:24]: 目前不区分。

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java [hunk 4]
```diff
@@ -166,5 +171,50 @@ public static Operator generate(TSLastDataQueryReq req, ZoneId zoneId)
     return queryOp;
   }
 
+  @TestOnly
+  public static Operator generate(String sql, ZoneId zoneId) throws ParseCancellationException {
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java:175]: 简化一下哈 = =

File: docs/UserGuide/Appendix/Keywords.md (hunk 2 above)
Comment (liuminghui233) [reply docs/UserGuide/Appendix/Keywords.md:22]: fixed!

File: docs/zh/UserGuide/IoTDB-SQL-Language/Syntax-Conventions.md [hunk 5]
```diff
@@ -21,103 +21,162 @@
...
+- `CREATE TRIGGER` 语句中描述触发器属性的键值对 
+- UDF 函数输入参数中的属性
+- `LOAD` / `REMOVE` / `SETTLE` 指令中的文件路径 
+- 用户密码
+
+通过以下几种方式可以在字符串内使用引号：
+
+- 使用两个连续单引号转义单引号，即 `''` 转义为 `'`。
+- 使用两个连续双引号转义双引号，即 `""` 转义为 `"`。
+- 在引号前使用转义符 (\\)。
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply docs/zh/UserGuide/IoTDB-SQL-Language/Syntax-Conventions.md:49]: fixed!

File: docs/zh/UserGuide/Appendix/Keywords.md (hunk 3 above)
Comment (liuminghui233) [reply docs/zh/UserGuide/Appendix/Keywords.md:24]: fixed!

File: server/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java (hunk 4 above)
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/qp/strategy/LogicalGenerator.java:175]: fixed!

File: integration/src/test/java/org/apache/iotdb/db/integration/IoTDBUDTFBuiltinFunctionIT.java [hunk 6]
```diff
@@ -201,7 +201,7 @@ public void testStringProcessingFunctions() {
         Statement statement = connection.createStatement()) {
       ResultSet resultSet =
           statement.executeQuery(
-              "select STRING_CONTAINS(s6, 's'='0'), STRING_MATCHES(s6, 'regex'='\\d') from root.sg.d1");
```
[the remark this answers is not in the document]
Comment (liuminghui233) [reply integration/src/test/java/org/apache/iotdb/db/integration/IoTDBUDTFBuiltinFunctionIT.java:204]: fixed!

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**